R/intervals.R

```diff
@@ -87,3 +87,28 @@ get_promoters <- function(upstream = 500, downstream = 50) {
         mutate(start = ifelse(strand == 1, start - upstream, start - downstream), end = ifelse(strand == 1, end + downstream, end + upstream)) %>%
         gintervals.force_range()
 }
+
+#' Convert 10X-format peak names to misha intervals
+#' @param pn peak names as generated by 10X ATAC pipeline ("{chr}-{start}-{end}" format)
+#'
+#' @export
+convert_10x_peak_names_to_misha_intervals = function(pn) {
+    intervals = as.data.frame(do.call('rbind', stringr::str_split(pn, '-')))
+    intervals[,2:3] = apply(intervals[,2:3], 2, as.numeric)
+    colnames(intervals) = c('chrom', 'start', 'end')
+    intervals$intervalID = 1:nrow(intervals)
```

Comment:
I do not think we need to add intervalID here - it is not a required field for misha intervals.

Reply:
I know but it is very convenient for filtering interval sets (I can change this to a "peak name" field, it will be a little less convenient than just a number, but will probably help to prevent mistakes).